dask_kubernetes/tests/test_async.py

```diff
@@ -390,6 +392,61 @@ async def test_constructor_parameters(k8s_cluster, pod_spec):
         assert pod.metadata.generate_name == "myname"
 
 
+@pytest.mark.asyncio
+async def test_passing_nodeport_host_to_scheduler_will_set_correct_host(
+    k8s_cluster, pod_spec
+):
+    cluster = KubeCluster(
+        pod_template=pod_spec,
+        namespace="default",
+        nodeport_host="10.10.10.10",
+        scheduler_service_type="NodePort",
+        **cluster_kwargs,
+    )
+
+    cluster.scheduler_pod_template = cluster._get_pod_template(
+        cluster.pod_template, pod_type="scheduler"
+    )
+    cluster.scheduler_pod_template.spec.containers[0].args = ["dask-scheduler"]
+    cluster.scheduler_pod_template = clean_pod_template(
+        pod_template=cluster.scheduler_pod_template, pod_type="scheduler"
+    )
+    await ClusterAuth.load_first(cluster.auth)
+    cluster._generate_name = escape(
+        cluster._generate_name.format(
+            user=getpass.getuser(), uuid=str(uuid.uuid4())[:10]
+        )
+    )
+    cluster.scheduler_pod_template = cluster._fill_pod_templates(
+        cluster.scheduler_pod_template, pod_type="scheduler"
+    )
+
+    scheduler = Scheduler(
+        idle_timeout=cluster._idle_timeout,
+        service_wait_timeout_s=cluster._scheduler_service_wait_timeout,
+        pod_template=cluster.scheduler_pod_template,
+        nodeport_host=cluster.nodeport_host,
+        scheduler_service_type=cluster._scheduler_service_type,
+        cluster=cluster,
+        core_api=kubernetes.client.CoreV1Api(),
+        policy_api=kubernetes.client.PolicyV1beta1Api(),
+        namespace=cluster.namespace,
+        loop=cluster.loop,
+    )
+
+    scheduler.service = await scheduler._create_service(
+        scheduler.kwargs["scheduler_service_type"]
+    )
+
+    external_address = await get_external_address_for_scheduler_service(
+        scheduler.core_api,
+        scheduler.service,
+        nodeport_host=scheduler.kwargs["nodeport_host"],
+    )
+    assert external_address == "tcp://10.10.10.10:8786"
+    assert scheduler.service.spec.type == "NodePort"
```

Comment:
I appreciate how in depth you've gone here with this test but given that this had gone so deep into the guts of the library I worry that it will be brittle and break if we make internal changes.

Perhaps it would be best to focus on testing the `get_external_address_for_scheduler_service` in isolation instead.

```suggestion
    from dask_kubernetes.objects import make_service_from_dict
    
    service = make_service_from_dict({"spec": {"ports": [{"port": 8786, "name": "comm"}], "type": "NodePort"}})
    
    external_address = await get_external_address_for_scheduler_service(
        None, service, nodeport_host="10.10.10.10",
    )
    assert external_address == "tcp://10.10.10.10:8786"
```

Reply:
Thanks! Couldn't figure out how to create an instance of the service without going through the cluster - `make_service_from_dict` was the missing piece :smile: